Source/HTTP/Curl/CurlMulti.cpp

```diff
@@ -94,16 +94,22 @@ HRESULT CALLBACK CurlMulti::CleanupAsyncProvider(XAsyncOp op, const XAsyncProvid
         assert(multi->m_cleanupAsyncBlock == nullptr);
         multi->m_cleanupAsyncBlock = data->async;
 
-        if (multi->m_taskQueueCallbacksPending == 0)
+        // If there are no pending task queue callbacks (and thus no pending HTTP requets), schedule cleanup immediately.
+        // If this condition is true, we're also guaranteed to be the final remaining reference to the CurlMulti.
+        bool cleanupImmediately = multi->m_taskQueueCallbacksPending == 0;
+        XTaskQueueHandle queue = multi->m_queue;
+
+        // Release the lock before going any further because the multi may be destroyed before Begin completes
+        lock.unlock();
+
```

Comment:
nit: it might not be a bad idea to null out the `multi` temporary after this unlocks or after the async work is scehduled. Just as a guard against future programmer misuse to not reference the muli-object after scheduling the async work.

Reply:
added.